tests/trinity/core/configuration/test_chain_config_object.py

```diff
@@ -0,0 +1,105 @@
+import pytest
+
+from eth_utils import (
+    decode_hex,
+    to_int,
+)
+
+from eth.db.atomic import AtomicDB
+from eth.vm.forks.constantinople import ConstantinopleVM
+from eth.vm.forks.homestead import HomesteadVM
+from eth.chains.mainnet import (
+    MainnetChain,
+    MAINNET_GENESIS_HEADER,
+)
+from eth.chains.ropsten import (
+    RopstenChain,
+    ROPSTEN_GENESIS_HEADER,
+)
+
+from trinity.config import (
+    ChainConfig,
+)
+from trinity.constants import (
+    MAINNET_NETWORK_ID,
+    ROPSTEN_NETWORK_ID,
+)
+from trinity.utils.db import MemoryDB
+from trinity.utils.eip1085 import validate_raw_eip1085_genesis_config
+
+
+def assert_vm_configuration_equal(left, right):
+    assert len(left) == len(right), "Length mismatch"
+
+    for ((left_block, left_vm), (right_block, right_vm)) in zip(left, right):
+        assert left_vm.fork is not None
+        assert left_vm.fork == right_vm.fork
+        assert left_block == right_block
+
+        if isinstance(left_vm, HomesteadVM):
+            assert left_vm.support_dao_fork is right_vm.support_dao_fork
+            assert left_vm.dao_fork_block_number == right_vm.dao_fork_block_number
+
+
+@pytest.mark.parametrize(
+    'network_id',
+    (MAINNET_NETWORK_ID, ROPSTEN_NETWORK_ID),
+)
+def test_chain_config_from_preconfigured_network(network_id):
+    chain_config = ChainConfig.from_preconfigured_network(network_id)
+    chain = chain_config.initialize_chain(AtomicDB(MemoryDB()))
+
+    if network_id == MAINNET_NETWORK_ID:
+        assert chain_config.chain_id == MainnetChain.chain_id
```

Comment:
I guess it's super trivial, but maybe compare network IDs too.

Reply:
The `network_id` does not exist on the `ChainConfig` and it's already doing `if network_id == MAINNET_NETWORK_ID`. Do I misread what you are suggesting?

Reply:
But there is no `network_id` on the `ChainConfig`. The `network_id` exists on the `TrinityConfig`.

Reply:
Yup, you're right, I was thinking of the TrinityConfig. Did that used to be called ChainConfig? Anyway, nothing to do here AFAICT.

Reply:
They got split into `ChainConfig` and `TrinityConfig`